lightly/cli/upload_cli.py

```diff
@@ -99,7 +103,7 @@ def _upload_cli(cfg, is_cli_call=True):
         print('Starting upload of embeddings.')
         try:
             embeddings = api_workflow_client.embeddings_api \
-                .get_embeddings_by_dataset_id(dataset_id=dataset_id)
+                .get_embeddings_by_dataset_id(dataset_id=api_workflow_client.dataset_id)
```

Comment:
is there a reason you didn't adopt nathan's idea of adding a return value to the `create_dataset` routine?

Reply:
Yes: if a user then creates a new dataset or switches to another dataset using the `api_workflow_client`, the `api_workflow_client.dataset_id` and `dataset_id` are again different. Thus I would prefer that always the `api_workflow_client.dataset_id`, as it is always up to date.

Reply:
Ok, but in the following workflow this could lead to problems:
- create a dataset X
- create a dataset Y
- I want to do something with the dataset id of X

It's clear that the returned `dataset_id` is static and hence easy to debug. However, it's not clear that the `api_workflow_client.dataset_id` is *not* static. So IMO it would be good to return the dataset id.

Reply:
If a user wants to work with two different datasets, then two different `ApiWorkflowClient`s must be created. Most methods of the `ApiWorkflowClient` or one of its `Mixins` use the self.dataset_id anyway.

I think it would be better to make all the apis within the `ApiWorkflowClient` private and thus disallow calling `api_workflow_client.embeddings_api.get_embeddings_by_dataset_id `. Instead, there should be a method `ApiWorkflowClient.get_embeddings()`, which does the same, but a) inserts the dataset_id itself and b) creates typehints

Reply:
Third Solution: put the whole code in
https://github.com/lightly-ai/lightly/blob/592d8f899a427dd0763b66ef856c8ffd74179357/lightly/cli/upload_cli.py#L99-L137
into 
https://github.com/lightly-ai/lightly/blob/592d8f899a427dd0763b66ef856c8ffd74179357/lightly/api/api_workflow_upload_embeddings.py#L71-L78

Reply:
I put this into another issue: https://github.com/lightly-ai/lightly/issues/568